src/platform/packages/shared/shared-ux/code_editor/impl/react_monaco_editor/editor.tsx

```diff
@@ -174,6 +179,11 @@ export function MonacoEditor({
     editorWillUnmount?.(editor.current!, monaco);
   };
 
+  useEffect(() => {
+    // initialize supported languages
+    initializeSupportedLanguages();
```

Comment:
would it make sense to initialize only requested language? I could imagine it would save some cpu ticks on initial render? 
Also I imagine if a language is already registered, calling this again is noop?

Reply:
I also wonder, maybe for later, how "large" are those language files? Would it make sense to load a language bundle when it is requested.

Reply:
This hook also at least missing `[]` an empty dependency array

Reply:
> would it make sense to initialize only requested language? I could imagine it would save some cpu ticks on initial render? Also I imagine if a language is already registered, calling this again is noop?

I did consider this, however since the languages definitions are already bundled within `@kbn/monaco` has a static import, I'm not sure if we'd see any gains with only registering the requested language. Ideally that's what I'd prefer too. To the question about calling registration again, yes it's configured to be a noop.

Reply:
> I also wonder, maybe for later, how "large" are those language files? Would it make sense to load a language bundle when it is requested.
> 
> Ok, I see this has to be implemented in onLanguage function. we could audit if there is any language that would benefit from better async chunk loading

You're right we might get some benefits from having the languages register async, I however wanted to have that in the next iteration of improvements for the code editor, so we might then to your point prevent imports from `@kbn/monaco` directly, however it would increase the surface area of this PR hence why I opted to not include that change here. I'd create an issue to track this

Reply:
Should at least fix now

> This hook also at least missing [] an empty dependency array

Reply:
> Should at least fix now
> 
> > This hook also at least missing [] an empty dependency array

typically ignoring the dependency array is acceptable, but it's also made me realize I picked the wrong tool for the job.

Reply:
So keeping useEffect approach? If so, I'd still add `[]`, because we do't need to run the effect on every render